docs/faq.md

```diff
@@ -0,0 +1,104 @@
+
+
+### Why use `BigInt` to represent 64-bit integers?
+
+The short answer is that they are the best way to represent the 64-bit numerical types allowable in Protobuf.  `BigInt` has [widespread browser support](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt#browser_compatibility) and for those environments where it is not supported, we fall back to a [string representation](https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md#bigint-in-unsupported-environments).  
+
+While it is true that an `int32`'s 2^32 size is not enough to represent a 64-bit value, Javascript's [`MAX_SAFE_INTEGER`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MAX_SAFE_INTEGER#description) can safely represent integers between -(2^53 – 1) and 2^53 – 1.  However, this is obviously only effective if you can guarantee that no number in that field will ever exceed this range.  This could lead to subtle and potentially serious bugs, so the clear-cut usage of `BigInt` makes more sense.
+
+### Why generate classes instead of interfaces?
+
+This is definitely something we considered.  We are aware of the debates on whether JavaScript classes should be used and whether they're actually worthwhile.  We chose to generate classes for a few reasons:
+
+- Protobuf messages and classes are very similar in how they represent data.  They are both encapsulating objects that contain properties that describe the overall entity.  A class is a great way to model this relationship.  The [official MDN documentation on classes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_Classes#why_classes) states:
+
+  > In general, you should consider using classes when you want to create objects that store their own internal data and expose a lot of behavior.
+
+  Which is exactly what our aim is when generating code from a Protobuf file.
+
+- Our messages contain numerous additional functions such as constructing an object, serializing to and from JSON, serializing to and from binary data, and cloning.  Most of this functionality is reusable, so using inheritance with a `Message` super class is beneficial.
+
+- We want the generated code to be approachable.  While the behavior of classes under the hood may be surprising as to how they're constructed and represented, they make for a very easy-to-read representation of a message.  The encapsulation they provide and some assurances that TypeScript provides on top of them make for a compelling argument to readability.
+
+So, in summary, yes, we know that JavaScript is not an OOP language and classes were shoehorned in, but our use case seems to be the reason they were added.
```

Comment:
Is JavaScript not object-oriented? It has first-class functions, but has _also_ had prototype-based object-orientation from the very start. To quote Douglas Crockford:

> What could be more object oriented than that?

It is a language that provides multiple paradigms. We picked the object-oriented approach to represent messages because it is a good fit.